### ISSUE TITLE
[BUG] crashing at launch with: PermissionError: [WinError 5] Access is denied: 'AppData' -> 'AppData_OLD'

### ISSUE THREAD

**Issue by baccccccc** [Jules-WinnfieldX/CyberDropDownloader#663]
**Describe the bug**
The recent version of the app crashes at startup. The previous (yesterday's) 5.x version did not crash.
I know it sounds lame, but I have no idea what was the last good version. Yesterday it worked, today I upgraded to the latest one, and now it crashes.
I also don't remember what the current version is. Because it crashes, I cannot look up any meaningful output.

**Important**
I am using [Python 3.12 from Microsoft Store](https://www.microsoft.com/store/productId/9NCVDN91XZQP). This is [an official distribution](https://docs.python.org/3/using/windows.html#windows-store) by Python software foundation. I do not have any other version of Python installed, and I don't want to.

There may be some limitations associated with how Python is packaged due to the Microsoft Store requirements. Also, some paths may be different. I am always using the full paths as you can see below. And everything worked this way before (both on 4.x and on 5.x branches.) This is the first major issue I am having in months of using this app under packaged Python.

**To Help Reproduce**
1. Install [Python 3.12 from Microsoft Store](https://www.microsoft.com/store/productId/9NCVDN91XZQP).
2. Open PowerShell. (Legacy command line will also work, although the environmental variable for user profile will be different, so you will have to adjust the below command.)
3. Install/upgrade CDL: `& "$env:localAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts\pip3.12.exe" install --upgrade 'Cyberdrop-dl'`
4. Attempt to launch CD: `& "$env:localAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts\cyberdrop-dl.exe"`
5. Observe the failure below.

**Screenshots**
```
PS C:\Users\<redacted>> & "$env:localAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts\pip3.12.exe" install --upgrade 'Cyberdrop-dl'
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: Cyberdrop-dl in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (5.0.34)
Requirement already satisfied: aiofiles==0.8.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (0.8.0)
Requirement already satisfied: aiohttp<4.0.0,>=3.9.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (3.9.1)
Requirement already satisfied: aiolimiter<2.0.0,>=1.1.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (1.1.0)
Requirement already satisfied: aiorun<2024.0.0,>=2023.7.2 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (2023.7.2)
Requirement already satisfied: aiosqlite==0.17.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (0.17.0)
Requirement already satisfied: asyncpraw<8.0.0,>=7.7.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (7.7.1)
Requirement already satisfied: beautifulsoup4<5.0.0,>=4.12.2 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (4.12.2)
Requirement already satisfied: browser-cookie3<0.20.0,>=0.19.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (0.19.1)
Requirement already satisfied: certifi<2024.0.0,>=2023.11.17 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (2023.11.17)
Requirement already satisfied: filedate<3.0,>=2.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (2.0)
Requirement already satisfied: get-video-properties<0.2.0,>=0.1.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (0.1.1)
Requirement already satisfied: inquirerpy<0.4.0,>=0.3.4 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (0.3.4)
Requirement already satisfied: mediafire<0.7.0,>=0.6.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (0.6.1)
Requirement already satisfied: mutagen<2.0.0,>=1.47.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (1.47.0)
Requirement already satisfied: myjdapi<2.0.0,>=1.1.7 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (1.1.7)
Requirement already satisfied: pillow<11.0.0,>=10.1.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (10.1.0)
Requirement already satisfied: platformdirs<4.0.0,>=3.11.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (3.11.0)
Requirement already satisfied: pyyaml<7.0.0,>=6.0.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (6.0.1)
Requirement already satisfied: rich<14.0.0,>=13.7.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from Cyberdrop-dl) (13.7.0)
Requirement already satisfied: typing_extensions>=3.7.2 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from aiosqlite==0.17.0->Cyberdrop-dl) (4.8.0)
Requirement already satisfied: attrs>=17.3.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from aiohttp<4.0.0,>=3.9.1->Cyberdrop-dl) (23.1.0)
Requirement already satisfied: multidict<7.0,>=4.5 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from aiohttp<4.0.0,>=3.9.1->Cyberdrop-dl) (6.0.4)
Requirement already satisfied: yarl<2.0,>=1.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from aiohttp<4.0.0,>=3.9.1->Cyberdrop-dl) (1.9.3)
Requirement already satisfied: frozenlist>=1.1.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from aiohttp<4.0.0,>=3.9.1->Cyberdrop-dl) (1.4.0)
Requirement already satisfied: aiosignal>=1.1.2 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from aiohttp<4.0.0,>=3.9.1->Cyberdrop-dl) (1.3.1)
Requirement already satisfied: asyncprawcore<3,>=2.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from asyncpraw<8.0.0,>=7.7.1->Cyberdrop-dl) (2.3.0)
Requirement already satisfied: update-checker>=0.18 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from asyncpraw<8.0.0,>=7.7.1->Cyberdrop-dl) (0.18.0)
Requirement already satisfied: soupsieve>1.2 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from beautifulsoup4<5.0.0,>=4.12.2->Cyberdrop-dl) (2.5)
Requirement already satisfied: lz4 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from browser-cookie3<0.20.0,>=0.19.1->Cyberdrop-dl) (4.3.2)
Requirement already satisfied: pycryptodomex in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from browser-cookie3<0.20.0,>=0.19.1->Cyberdrop-dl) (3.19.0)
Requirement already satisfied: python-dateutil in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from filedate<3.0,>=2.0->Cyberdrop-dl) (2.8.2)
Requirement already satisfied: pfzy<0.4.0,>=0.3.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from inquirerpy<0.4.0,>=0.3.4->Cyberdrop-dl) (0.3.4)
Requirement already satisfied: prompt-toolkit<4.0.0,>=3.0.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from inquirerpy<0.4.0,>=0.3.4->Cyberdrop-dl) (3.0.41)
Requirement already satisfied: requests in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from mediafire<0.7.0,>=0.6.1->Cyberdrop-dl) (2.31.0)
Requirement already satisfied: requests-toolbelt in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from mediafire<0.7.0,>=0.6.1->Cyberdrop-dl) (1.0.0)
Requirement already satisfied: six in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from mediafire<0.7.0,>=0.6.1->Cyberdrop-dl) (1.16.0)
Requirement already satisfied: pycryptodome in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from myjdapi<2.0.0,>=1.1.7->Cyberdrop-dl) (3.19.0)
Requirement already satisfied: markdown-it-py>=2.2.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from rich<14.0.0,>=13.7.0->Cyberdrop-dl) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from rich<14.0.0,>=13.7.0->Cyberdrop-dl) (2.17.2)
Requirement already satisfied: mdurl~=0.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from markdown-it-py>=2.2.0->rich<14.0.0,>=13.7.0->Cyberdrop-dl) (0.1.2)
Requirement already satisfied: wcwidth in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from prompt-toolkit<4.0.0,>=3.0.1->inquirerpy<0.4.0,>=0.3.4->Cyberdrop-dl) (0.2.12)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from requests->mediafire<0.7.0,>=0.6.1->Cyberdrop-dl) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from requests->mediafire<0.7.0,>=0.6.1->Cyberdrop-dl) (3.4)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\<redacted>\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages (from requests->mediafire<0.7.0,>=0.6.1->Cyberdrop-dl) (2.1.0)
PS C:\Users\<redacted>> & "$env:localAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts\cyberdrop-dl.exe"
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts\cyberdrop-dl.exe\__main__.py", line 7, in <module>
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\cyberdrop_dl\main.py", line 103, in main
    manager = startup()
              ^^^^^^^^^
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\cyberdrop_dl\main.py", line 26, in startup
    manager = Manager()
              ^^^^^^^^^
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\cyberdrop_dl\managers\manager.py", line 37, in __init__
    self.first_time_setup.startup()
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\cyberdrop_dl\utils\transfer\first_time_setup.py", line 39, in startup
    APP_STORAGE.rename(APP_STORAGE.with_name("AppData_OLD"))
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.240.0_x64__qbz5n2kfra8p0\Lib\pathlib.py", line 1364, in rename
    os.rename(self, target)
PermissionError: [WinError 5] Access is denied: 'AppData' -> 'AppData_OLD'
PS C:\Users\<redacted>>
```

**Setup Information**
 - OS: x64 Windows 11, version 23H2, latest version with all updates from Widnows Update

**Comment by Jules-WinnfieldX** [comment]
Wild that you're using the windows store install successfully at all. But we'll skip that. 

The latest release of CDL has moved the files in appdata back to the location that you're running CDL from. 

I'm unsure of your setup, but where are you the commands from?

**Comment by Aliaksandr3by** [comment]
cyberdrop-dl==5.0.24 have messed up with folder \AppData\

**Comment by Jules-WinnfieldX** [comment]
Also, before we get into the rigamorole. If you want to leave the files in the appdata directory, let me know and we can sort that for you.

**Comment by Jules-WinnfieldX** [comment]
If you want, you can message me on discord and I can try and help there.

**Comment by baccccccc** [comment]
> Wild that you're using the windows store install successfully at all.
I know, right? But someone gotta test this shit! And I'm a fan of package managers in general.

> The latest release of CDL has moved the files in appdata back to the location that you're running CDL from.
ahh, interesting. Well, so far, I was trying to run it from my profile root (`$env:UserProfile` aka `C:\Users\<username>`). Normally I run it from the folder with downloads. But this time, I was only going to change some configs. So I ran it from the default path.

Now that I better understand what's going on, I've tried to change location to where I normally download stuff.

And I got a different error:

```
PS C:\Users\<redacted>> Set-Location -Path 'C:\Mirror — Temp\cdl'
PS C:\Mirror — Temp\cdl> & "$env:localAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts\cyberdrop-dl.exe"
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts\cyberdrop-dl.exe\__main__.py", line 7, in <module>
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\cyberdrop_dl\main.py", line 103, in main
    manager = startup()
              ^^^^^^^^^
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\cyberdrop_dl\main.py", line 26, in startup
    manager = Manager()
              ^^^^^^^^^
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\cyberdrop_dl\managers\manager.py", line 37, in __init__
    self.first_time_setup.startup()
  File "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\cyberdrop_dl\utils\transfer\first_time_setup.py", line 40, in startup
    OLD_APP_STORAGE.rename(APP_STORAGE)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.240.0_x64__qbz5n2kfra8p0\Lib\pathlib.py", line 1364, in rename
    os.rename(self, target)
OSError: [WinError 17] The system cannot move the file to a different disk drive: 'C:\\Users\\<redacted>\\AppData\\Local\\Cyberdrop-DL\\Cyberdrop-DL' -> 'AppData'
PS C:\Mirror — Temp\cdl>
```

This was actually quite helpful because I can totally move this folder myself, I guess.

Except that the folder is actually not there. Because of how the Store version of Python is packaged, the folder is actually here:

`"$env:LocalAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL"`

So, now you tell me. Do I need to move only some files from this folder, or all files? And where exactly do I move them? To the root of my download folder or to some subfolder?

**Comment by Jules-WinnfieldX** [comment]
Assuming you have an AppData folder in the location you specified: `"$env:LocalAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL"`

You'll want to move it to wherever you are calling the script from. So If you want to start it in `C:\Mirror — Temp\cdl` move it there. You'll need to run the commands from that directory. The paths internally are set to relative paths. ./AppData and ./Downloads (obviously downloads can be remapped, appdata cannot currently)

**Comment by baccccccc** [comment]
Gotcha!

Ok, I tried to move all the files:

```
PS C:\Mirror — Temp\cdl> Get-ChildItem -Path "$env:LocalAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL" | Move-Item -Destination 'C:\Mirror — Temp\cdl' -Verbose
VERBOSE: Performing the operation "Move Directory" on target "Item: C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL\Cache Destination: C:\Mirror — Temp\cdl\Cache".
VERBOSE: Performing the operation "Create Directory" on target "Destination: C:\Mirror — Temp\cdl\Cache".
VERBOSE: Performing the operation "Remove Directory" on target "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL\Cache".
VERBOSE: Performing the operation "Move Directory" on target "Item: C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL\Configs Destination: C:\Mirror — Temp\cdl\Configs".
VERBOSE: Performing the operation "Create Directory" on target "Destination: C:\Mirror — Temp\cdl\Configs".
VERBOSE: Performing the operation "Create Directory" on target "Destination: C:\Mirror — Temp\cdl\Configs\Default".
VERBOSE: Performing the operation "Create Directory" on target "Destination: C:\Mirror — Temp\cdl\Configs\Default\Logs".
VERBOSE: Performing the operation "Remove Directory" on target "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL\Configs\Default\Logs".
VERBOSE: Performing the operation "Remove Directory" on target "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL\Configs\Default".
VERBOSE: Performing the operation "Remove Directory" on target "C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL\Configs".
VERBOSE: Performing the operation "Move File" on target "Item: C:\Users\<redacted>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL\download_history.sqlite Destination: C:\Mirror — Temp\cdl\download_history.sqlite".
```

But I kept receiving the same errors.

So removed the folder completely (now that it was empty)

```
Remove-Item -Path "$env:LocalAppData\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\Local\Cyberdrop-DL\Cyberdrop-DL"
```

Now the app seems to launch correctly!

Although I'm not sure I did everythign right because it looks like all of my settings are gone :)

But no big deal, I can set them back now.

Thanks for help!

**Comment by Jules-WinnfieldX** [comment]
Of course, sorry for the roller coaster.

**Comment by baccccccc** [comment]
Ok, I fugred why my previous settings did not take effect. I moved everything to the root of `C:\Mirror — Temp\cdl`. (This is my current directory when I run the app now. Previously, I used to run it from sub-folders such as `C:\Mirror — Temp\cdl\ForumThreadAboutXYZ` because I wanted completely separate log and error files.)

So, now the config file for example is sitting here: `"C:\Mirror — Temp\cdl\Configs\Default\settings.yaml"`

But in fact, the app looks inside an `AppData` folder nested under the current path. So that I should have placed my old config into `"C:\Mirror — Temp\cdl\AppData\Configs\Default\settings.yaml`

I have moved stuff accordingly, and now the old settings are back!

P.S. there's also a new folder `C:\Mirror — Temp\cdl\Old Files` and I'm not quite sure what to do with it. I guess I'll just keep it around for a while, and then delete.
